src/converse-register.js

```diff
@@ -241,6 +245,25 @@
                     return false;
                 },
 
+                chooseProvider: function (ev) {
```

Comment:
`chooseProvider` and `onProviderChosen` are both very similar. I think they can be combined into one method instead (keep the name `onProviderChosen`).